From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Greg Price <greg@zulip.com>
Date: Tue, 24 May 2022 23:00:44 -0700
Subject: =?UTF-8?q?[upstream]=20rnav:=20On=20create=E2=80=A6Navigator,=20c?=
 =?UTF-8?q?ut=20redundant=20wrong=20version=20of=20DefaultNav=E2=80=A6Opti?=
 =?UTF-8?q?ons?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This is already taken care of by TypedNavigator.  (The only thing this
Props type is used for is the parameter to the unexported function on
the next line; the only thing the latter is used for is as an argument
to TypedNavigator on the line after that.)

And TypedNavigator actually passes the ParamList type when it does so,
so it has a chance of being right.
---
 .../src/navigators/createBottomTabNavigator.js.flow      | 9 +--------
 .../src/navigators/createMaterialTopTabNavigator.js.flow | 9 +--------
 .../src/navigators/createStackNavigator.js.flow          | 6 +++---
 3 files changed, 5 insertions(+), 19 deletions(-)

diff --git types/@react-navigation/bottom-tabs/lib/typescript/src/navigators/createBottomTabNavigator.js.flow types/@react-navigation/bottom-tabs/lib/typescript/src/navigators/createBottomTabNavigator.js.flow
index 7e16fb708..ebf13d3dc 100644
--- types/@react-navigation/bottom-tabs/lib/typescript/src/navigators/createBottomTabNavigator.js.flow
+++ types/@react-navigation/bottom-tabs/lib/typescript/src/navigators/createBottomTabNavigator.js.flow
@@ -17,14 +17,7 @@ import {
   type BottomTabNavigationOptions,
 } from '../types';
 
-type Props = DefaultNavigatorOptions<
-  ParamListBase,
-  TabNavigationState<ParamListBase>,
-  BottomTabNavigationOptions,
-  BottomTabNavigationEventMap,
-> &
-  TabRouterOptions &
-  BottomTabNavigationConfig;
+type Props = TabRouterOptions & BottomTabNavigationConfig;
 declare function BottomTabNavigator(Props): $tsflower_subst$React$JSX$Element;
 declare var _default: <
   ParamList: ParamListBase,
diff --git types/@react-navigation/material-top-tabs/lib/typescript/src/navigators/createMaterialTopTabNavigator.js.flow types/@react-navigation/material-top-tabs/lib/typescript/src/navigators/createMaterialTopTabNavigator.js.flow
index bc11a7c76..f0a91a5ad 100644
--- types/@react-navigation/material-top-tabs/lib/typescript/src/navigators/createMaterialTopTabNavigator.js.flow
+++ types/@react-navigation/material-top-tabs/lib/typescript/src/navigators/createMaterialTopTabNavigator.js.flow
@@ -17,14 +17,7 @@ import {
   type MaterialTopTabNavigationOptions,
 } from '../types';
 
-type Props = DefaultNavigatorOptions<
-  ParamListBase,
-  TabNavigationState<ParamListBase>,
-  MaterialTopTabNavigationOptions,
-  MaterialTopTabNavigationEventMap,
-> &
-  TabRouterOptions &
-  MaterialTopTabNavigationConfig;
+type Props = TabRouterOptions & MaterialTopTabNavigationConfig;
 declare function MaterialTopTabNavigator(Props): $tsflower_subst$React$JSX$Element;
 declare var _default: <
   ParamList: ParamListBase,
diff --git types/@react-navigation/stack/lib/typescript/src/navigators/createStackNavigator.js.flow types/@react-navigation/stack/lib/typescript/src/navigators/createStackNavigator.js.flow
index 3c2939d32..f96ddf688 100644
--- types/@react-navigation/stack/lib/typescript/src/navigators/createStackNavigator.js.flow
+++ types/@react-navigation/stack/lib/typescript/src/navigators/createStackNavigator.js.flow
@@ -16,9 +16,9 @@ import {
   type StackNavigationEventMap,
 } from '../types';
 
-type Props = DefaultNavigatorOptions<StackNavigationOptions> &
-  StackRouterOptions &
-  StackNavigationConfig;
+type Props =
+  // DefaultNavigatorOptions<StackNavigationOptions> &
+  StackRouterOptions & StackNavigationConfig;
 declare function StackNavigator(Props): $tsflower_subst$React$JSX$Element;
 declare var _default: <
   ParamList: { +[key: string]: { ... } | void },
-- 
2.32.0

